edn/src/types.rs

```diff
@@ -143,6 +162,16 @@ impl Value {
     def_as!(as_list, List, LinkedList<Value>);
     def_as!(as_set, Set, BTreeSet<Value>);
     def_as!(as_map, Map, BTreeMap<Value, Value>);
+
+    pub fn from_bigint(src: &str) -> Result<Value, NewValueError> {
```

Comment:
In my experience, detailed error reporting is often a waste of effort: it's only worthwhile when the caller's behavior needs to change based on what went wrong.

In this case, the caller should almost always have a valid string — it parsed it from user input using a parser that includes bigint syntax — and the only error state we need to communicate is "we couldn't turn this into a bigint for some reason". The caller already has the string and can use that for error reporting if it chooses. The user will probably be able to see what's wrong without more help.

The other scenario to consider is that often it's *not* failure *per se*; it might just have been a speculative attempt that wasn't the right branch. Parsing is often phrased in terms of "is it an X, a Y, or a Z?". With `satisfy_map`, the parsing returns an `Option<T>` to indicate that this branch of the parse didn't succeed. The error message presented to the user would then be something like "expected X, Y, or Z; got <input>". A detailed error from `parse_x` is of no use, and would be discarded immediately.

A final thought: one reason for `Result` is that it requires callers to check the value. That implies it's used for side-effecting things like filesystem writes; to me it feels more like something where the error is important, even primary.